packages/teleport-generator-shared/src/utils/project-utils.ts

```diff
@@ -248,14 +249,144 @@ export const createFile = (name: string, fileType: string, content: string): Gen
   return { name, fileType, content }
 }
 
-export const createFolder = (
-  name: string,
-  files: GeneratedFile[] = [],
-  subFolders: GeneratedFolder[] = []
+interface LocalDependenciesMeta {
+  defaultComponentsPath: string | string[]
+  defaultPagesPath: string | string[]
+}
+
+export const generateLocalDependenciesPrefix = (
+  template: TemplateDefinition,
+  meta: LocalDependenciesMeta
+): string => {
+  const initialComponentsPath =
+    template.meta && template.meta.componentsPath
+      ? template.meta.componentsPath
+      : [].concat(meta.defaultComponentsPath)
+
+  const initialPagesPath =
+    template.meta && template.meta.pagesPath
+      ? template.meta.pagesPath
+      : [].concat(meta.defaultPagesPath)
+
+  let dependencyPrefix = ''
+
+  /*
+    Remove common path elements from the beginning of the
+    components and pages full path (if any)
+  
+    For example, having:
+    - initialComponentsPath = ['src', 'components']
+    - initialPagesPath = ['src', 'pages']
+  
+    If we want to have an import statement that goes from the pages folder to the
+    components folder, we only need to go back one step, so we are removing
+    the forst element from both the paths ('src') and build the dependencyPrefix accordingly
+  */
+  const { componentsPath, pagesPath } = removeCommonStartingPointsFromPaths(
+    initialComponentsPath,
+    initialPagesPath
+  )
+
+  // We have to go back as many folders as there are defined in the pages path
+  dependencyPrefix += '../'.repeat(pagesPath.length)
+
+  dependencyPrefix += componentsPath
+    .map((component) => {
+      return `${component}/`
+    })
+    .join('')
+
+  return dependencyPrefix
+}
+
+const removeCommonStartingPointsFromPaths = (
```

Comment:
can this function work on any two concepts / folders? or is it only for pages and components?

Reply:
I've made this function more generic. It now accepts an array of path arrays and removes the common starting points from all of them